core/capabilities/compute/compute.go

```diff
@@ -329,6 +336,14 @@ func (c *Compute) createFetcher() func(ctx context.Context, req *wasmpb.FetchReq
 			return nil, fmt.Errorf("failed to unmarshal fetch response: %w", err)
 		}
 
+		c.metrics.with(
```

Comment:
do we also need to add the refID? If so, I'll need to make some changes in common to add it as metadata. @cedric-cordenier

Reply:
for metrics, the execution ID and step ref are high cardinality and we are not using them as labels as a result. The rest we can keep, though in the future we may decide to remove WorkflowID as well.